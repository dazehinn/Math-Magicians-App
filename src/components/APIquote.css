p {
  text-decoration: none;
  font-weight: lighter;
}
