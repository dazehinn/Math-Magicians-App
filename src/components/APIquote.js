import { useState, useEffect } from 'react';
import './APIquote.css';
import { v4 as uuidv4 } from 'uuid';

const APIquote = () => {
  const myuuid = uuidv4();
  const [data, setData] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [hasError, sethasError] = useState(null);

  useEffect(() => {
    const fetchData = async () => {
      const url = 'https://api.api-ninjas.com/v1/quotes?category=life';
      try {
        setIsLoading(true);
        const res = await fetch(url, {
          method: 'GET',
          headers: { 'X-Api-Key': 'rqBsI0z9wOGAqafKnCkkEA==KMMCPBuS2rX5v54e' },
          contentType: 'application/json',
        });
        const response = await res.json();
        setData(response);
      } catch (error) {
        sethasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [setData, setIsLoading]);

  if (hasError) return <div>An error occured while loading data</div>;

  if (isLoading) return <div>Loading...</div>;

  return (
    <ul>
      {data.map((item) => (
        <>

{data.map((item) => (
        <li key={myuuid}>
          <p>{item.quote}</p>
          <p>
            <i>{item.author}</i>
          </p>
        </li>
))}
        </>
      ))}
    </ul>
  );
};

export default APIquote;
