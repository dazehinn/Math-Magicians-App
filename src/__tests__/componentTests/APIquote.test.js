import renderer from 'react-test-renderer';
import APIquote from '../../components/APIquote';

describe('Quote Component', () => {
  test('Tree', () => {
    const tree = renderer.create(<APIquote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
