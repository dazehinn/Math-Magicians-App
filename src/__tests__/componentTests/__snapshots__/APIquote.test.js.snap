// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Component Tree 1`] = `<ul />`;
